src/openvic-simulation/map/Map.cpp

```diff
@@ -562,15 +562,28 @@ bool Map::load_province_positions(BuildingTypeManager const& building_type_manag
 	})(root);
 }
 
-bool Map::load_region_file(ast::NodeCPtr root) {
+bool Map::load_region_colours(ast::NodeCPtr root, std::vector<colour_t>& colours) {
+	return expect_dictionary(
+		[&colours](std::string_view key, ast::NodeCPtr value) -> bool {
+			if (key != "color") {
+				return false;
+			}
+			colour_t colour = colour_t::null();
+			const bool ret = expect_colour(assign_variable_callback(colour))(value);
+			colours.push_back(colour);
+			return ret;
```

Comment:
```suggestion
			return expect_colour(vector_callback(colours))(value);
```
No need to use a temporary colour variable, `vector_callback` can do this for you automatically